clippy_lints/src/trait_bounds.rs

```diff
@@ -70,6 +70,35 @@ declare_clippy_lint! {
     "Check if the same trait bounds are specified twice during a function declaration"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for multiple instances of the same where clause or trait bound
+    ///
+    /// ### Why is this bad?
+    /// Adds to code noise
+    ///
+    /// ### Example
+    /// ```rust
+    /// fn foo<T: Default + Default>(bar: T) {}
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// fn foo<T: Default>(bar: T) {}
+    /// ```
+    ///
+    /// ```rust
+    /// fn foo<T>(bar: T) where T: Default + Default {}
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// fn foo<T>(bar: T) where T: Default {}
+    /// ```
+    #[clippy::version = "1.62.0"]
+    pub REPEATED_WHERE_CLAUSE_OR_TRAIT_BOUND,
```

Comment:
Could this lint be included in the `TRAIT_DUPLICATION_IN_BOUNDS`? When thinking about a good lint name for this lint I thought `TRAIT_DUPLICATION_IN_BOUNDS` would actually the best name for this lint.

If this lint should be kept its own separate lint, the lint name should be

```suggestion
    pub REPEATED_WHERE_CLAUSES_OR_TRAIT_BOUNDS,
```

Reply:
I think it is possible to merge the two. 

I have a proposal since merging `trait_duplication_in_bounds` and `repeated_where_clause_or_trait_bound` in could be too much work and might mean the PR stalls. 

I propose we:
1. Merge this PR when it is acceptable without adding any more functionality.
2. Fix https://github.com/rust-lang/rust-clippy/issues/8757 and merge PR. I've been working on a fix that is based on some work in this PR to make comparable trait refs. This will also make the two more similar and easier to refactor.
3. I'll work on a refactor PR to merge `trait_duplication_in_bounds` and `repeated_where_clause_or_trait_bound`.

Reply:
@flip1995 what do you think of this proposal?